cubitpy/mesh_creation_functions.py

```diff
@@ -194,11 +208,20 @@ def extrude_mesh_normal_to_surface(
                 # Check that the normal and position are equal.
                 other_coordinates = node_id_pos_normal_map[node_id][0]
                 other_normal = node_id_pos_normal_map[node_id][1]
-                if (
-                    np.linalg.norm(my_coordinates - other_coordinates) > 1e-10
-                    or (np.linalg.norm(my_normal - other_normal)) > 1e-10
-                ):
-                    raise ValueError("Positions or normals do not match!")
+                if np.linalg.norm(my_coordinates - other_coordinates) < tol_coord:
+                    if (np.linalg.norm(my_normal - other_normal)) > tol_normal:
+                        if average_normals == True:
+                            my_normal = (other_normal + my_normal) / 2
```

Comment:
I am not sure if this code does anything. Is `my_normal` even used if we end up in this `if` branch?

Also if you average it like this, it will in general not be a unit normal vector any more.

Reply:
yes ,it worked somehow, but not in the desired way.